### DISCUSSION TITLE
Advice for best Server Setup for CPSat problems

### DISCUSSION THREAD

**Discussion by jweizman** [google/or-tools#2654] [CP-SAT questions]
hi

My model takes 60s to be solved.
Just took a 64 cores, 128Mb memory server and the result is not really different.
What could be done to improve the solving time?

As a side note, the reason why i ask is because our users have to make a few tries to get to an acceptable solution (for them)
That's why i'm looing to reduce the solve time
tx!
Jonatahn

**Comment by lperron** [comment]
just making sure, you are setting num_search_workers to 64 ?

**Comment by jweizman** [comment]
Yes

Le lun. 12 juil. 2021 à 10:33, Laurent Perron ***@***.***> a
écrit :

**Comment by jweizman** [comment]
Sorry i answered too fast :

solver.StringParameters = $"num_search_workers:64, log_search_progress:
true, max_time_in_seconds:60";

Jonathan Weizman


Le lun. 12 juil. 2021 à 10:40, Jonathan Weizman ***@***.***> a
écrit :

**Comment by lperron** [comment]
The next best thing is actually trying to improve the model.
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00

**Comment by jweizman** [comment]
any hint on which direction i should search?
Jonathan Weizman


Le lun. 12 juil. 2021 à 11:13, Laurent Perron ***@***.***> a
écrit :

> The next best thing is actually trying to improve the model.
> Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
> 00
>
>
>

**Reply by gregy4** [reply]
More info is needed … are you searching for optimal or feasible solution ?
In case you spend a lot of time searching for first solution, usage of hints can help. Exists some simpler variant of your model with less variables and constraints ? Usage of boolean variables instead integer variables is preferred.
